woocommerce/google-listings-and-ads
# Multichannel Marketing Integration

### Changes proposed in this Pull Request:



Related to https://github.com/woocommerce/woocommerce/issues/34548.

This PR implements an integration with the Multichannel Marketing library that we're building in WooCommerce core. The MCM library is still in development (https://github.com/woocommerce/woocommerce/pull/35099) and is not merged into WC core yet. I've chosen GLA as the first test subject since I had the most experience with and the MCM design (pe2C5g-1g-p2) was made with GLA in mind.

### Detailed test instructions:


Note: To test this PR, you need to checkout and build the [feature/34549-multichannel-marketing-framework](https://github.com/woocommerce/woocommerce/tree/feature/34549-multichannel-marketing-framework) branch of WC core. (This is why the automated unit tests are failing in GitHub)

1. Go to `WooCommerce Settings > Advanced > Features` and turn on the multichannel marketing feature.
2. Go to the Marketing > Overview page and confirm Google Listings & Ads appears in the "Installed marketing extensions" section
3. Ensure you have **NOT** set up the extension or completed onboarding. If you have, reset the extension data.
4. Confirm that a link to "Finish setup" is displayed next to each extension name if it has not been set up and links to their onboarding/setup process.
5. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with the "Settings" link and points to the extension's settings page.

Extra test to confirm that the integration will be disabled if WooCommerce is on a previous version:

- Checkout and build the `trunk` branch of WooCommerce core and confirm that the `GLAChannel` class is not registered. You can check this by either setting a xDebug break point on the class construct, or simply changing the value returned by `get_name` to something else and confirming that the name displayed in "Installed Extensions" is different.

### Additional details:
- The GLAChannel class will only be registered if the `WC_MCM_EXISTS` constant is previously defined by the WooCommerce core. This is to ensure there are no errors with previous versions of WC that don't include the required interface/classes.


### Changelog entry

> Add - Integration with WooCommerce Multichannel Marketing
